### DISCUSSION TITLE
Applying dR cut

### DISCUSSION THREAD

**Discussion by ico1036** [scikit-hep/coffea#497] [Q&A] [answered]
Hi all,

I try to do following selection using Coffea.

Photon = [ [ pho1 ], [ pho2, pho3 ], [ ] ]
Electron = [ [ e1, e2 ], [ e3, e4 ] , [e5] ]
Select photons with dR (photon, all Electron in the event) > 0.5

I try following method.
```python
pho_ele_pair` = `ak.argcartesian([Photon,Electron])
```  
>[ (0, 0), (0, 1) ] , [ (1,2 ) (1,3), (2,1), (2,3) ],  [ ] ]

```python
pho,ele = ak.unzip(pho_ele_pair)
```

pho > [ [0,0], [1,1,2,2], [ ] ] 
ele  > [ [0,1], [2,3,2,3], [ ] ]

```python
pho.delta_r(ele) > 0.5
```
>[ [T ,F], [T,T,T,F], [ ] ]

The result should be
```python
pho_sel_mask = [ [F], [T, F] , [ ] ]
```

but I can't find solution.
is there any clever method? 
Thank you in advance.

**Comment by nsmith-** [comment, marked as answer]
The missing ingredient is the `nested` option to `ak.cartesian`, e.g.
```python
pair = ak.cartesian({"pho": events.Photon, "ele": events.Electron}, nested=True)
pho_sel_mask = ak.all(pair.pho.delta_r(pair.ele) > 0.5, axis=-1)
```

Since this is such a common operation, NanoEvents has a shortcut method [metric_table](https://coffeateam.github.io/coffea/api/coffea.nanoevents.methods.vector.LorentzVector.html?highlight=nearest#coffea.nanoevents.methods.vector.LorentzVector.metric_table), which could be used as:
```python
pho_sel_mask = ak.all(events.Photon.metric_table(events.Electron) > 0.5, axis=-1)
```
(the default metric is `delta_r`)

**Reply by ico1036** [reply]
Thank you very much!
